src/Tools/dotnet-gcdump/DotNetHeapDump/EventPipeDotNetHeapDumper.cs

```diff
@@ -87,9 +87,10 @@ public static bool DumpFromEventPipe(CancellationToken ct, int processID, Memory
 
                     eventPipeDataPresent = true;
 
-                    if (gcNum < 0 && data.Depth == 2 && data.Type != GCType.BackgroundGC)
+                    if (gcNum < 0 && data.Depth == 2 && data.Type != GCType.BackgroundGC && data.Reason == GCReason.Induced)
                     {
                         gcNum = data.Count;
+                        memoryGraph.Is64Bit = data.PointerSize == 8;
```

Comment:
This line also looks like a new fix? If this is another piece of code that was missed in the initial port then same request as on the edit above - if it is important to be here then lets describe why it is being added in the PR description.

Reply:
I reverted these changes but hadn't pushed. The description for what remained I already have in the description.

Reply:
Sounds good : )